### ISSUE TITLE
🛬 [INBOUND] - drf-excel

### ISSUE THREAD

**Issue by FlipperPA** [django-commons/membership#45]
### What's the URL to the project?

https://github.com/wharton/drf-excel

### Who are the current maintainers?

Timothy Allen, Mathieu Rampant

### Does the project have the required attributes?

- [x] Does the project have tests?
- [X] Does the project have documentation?
- [X] Has the project adopted Django Commons's Code of Conduct?

### Who will be the new maintainers of the project?

We've got to wrap up a rudimentary test suite.

**Comment by tim-schilling** [comment]
Linking the drf-excel's issue for posterity. https://github.com/wharton/drf-excel/issues/86 Thank you Tim for opening this!

**Comment by FlipperPA** [comment]
Thanks to @browniebroke, we now have some rudimentary tests in place, @tim-schilling. Happy to transfer in when you are ready.

**Comment by tim-schilling** [comment]
@FlipperPA can you confirm that this is the [only test in the project](https://github.com/wharton/drf-excel/blob/main/tests/test_utilities.py#L4)? I've been trying to find others, but it seems like the work was to get a test harness up and running with CI. If I'm wrong, potentially ignore the rest of my comment.

If that's the case, the concern there is a new person can't come in and change things confidently. Maybe a decent middleground here is to write up two issues, 1 to add code coverage and 2 identify the main components that are critical to having tests. I wonder if we can pin an issue in a repo to help encourage efforts toward them.

I think if two of the other @django-commons/admins are good with that (and there aren't any strong arguments against), we can bring it in.

I don't want to slow things down unnecessarily, but I do feel like tests need to be prioritized for drf-excel to help others contribute.

**Comment by FlipperPA** [comment]
@tim-schilling Thanks to the efforts of @browniebroke, there's now a more robust test for the full XLSX workbook generation routine. https://github.com/wharton/drf-excel/blob/main/tests/test_viewset_mixin.py

**Comment by cunla** [comment]
@FlipperPA, I agree with @tim-schilling.

You can add coverage visibility to in the test github workflow @browniebroke implemented (See example [here](https://github.com/django-commons/django-tasks-scheduler/blob/master/.github/actions/test-coverage/action.yml) for publishing a coverage report as a comment in the PR).

Once there is coverage visibility, it should be easy to identify what components require more robust testing.

**Comment by browniebroke** [comment]
> See example [here](https://github.com/django-commons/django-tasks-scheduler/blob/master/.github/actions/test-coverage/action.yml?rgh-link-date=2024-10-04T14%3A53%3A53Z) for publishing a coverage report as a comment in the PR).

I don't really understand where the coverage is displayed with the solution, I looked at recent PRs, but they don't have a comment displaying coverage.

Regarding `drf-excel`, the coverage is currently displayed in the [logs of GitHub actions](https://github.com/wharton/drf-excel/actions/runs/11106858262/job/30856116357#step:5:37) (just in case you missed it).

On the other hand, I see that django-typer uses codecov, which is much easier to setup. I gave this a go: https://github.com/wharton/drf-excel/pull/90 I also have a follow up piece where I'm bumping coverage a bit: https://github.com/browniebroke/drf-excel/pull/3

**Comment by tim-schilling** [comment]
I did miss that. I'm good with this being merged into the org.

**Comment by FlipperPA** [comment]
@tim-schilling How does the actual merge happen? Do I just move the repo myself? Thank you!

**Comment by browniebroke** [comment]
I think that's documented here? https://github.com/django-commons/controls#new-project-playbook

PS: I've joined the drf-excel maintainance team since this issue was opened

**Comment by tim-schilling** [comment]
@FlipperPA @browniebroke one of the admins will reach out to schedule a time to do the transfer. Right now our transfers are clocking in at about 1.25 hours.

One thing you can do ahead of time that will help is answering the following questions:

-  Does the project have a test.pypi.org project?
-  Would you like to test your deployments with uploads to test.pypi.org (you'll need to create the project first)?
-  Would you like to use GitHub's merge commits?
-  Would you like to use GitHub's rebase and merge?
-  Would you like to use GitHub's squash and merge?
-  Would you like to use GitHub's Discussions?
-  Would you like to use GitHub's Wiki?

**Comment by cunla** [comment]
Hi, please use this link to book some time with me to do the transfer: https://cal.com/daniel-moran/dc-meeting.

Can we do [django-click](https://github.com/django-commons/membership/issues/44) at the same time or are we still waiting for GaretJax?

**Comment by browniebroke** [comment]
I can't speak for @FlipperPA but here are my 2 cents:

> * Does the project have a test.pypi.org project?

Not at the moment, no

> * Would you like to test your deployments with uploads to test.pypi.org (you'll need to create the project first)?

No preference for me, perhaps let's set it up if it means more consistency with other projects in the org? 

> * Would you like to use GitHub's merge commits?

Not my preferred way of merging, but I don't mind being enabled

> * Would you like to use GitHub's rebase and merge?

Yes

> * Would you like to use GitHub's squash and merge?

It would my personal preference, yes

> * Would you like to use GitHub's Discussions?

It's disabled at the moment, but could be useful, I think

> * Would you like to use GitHub's Wiki?

No

**Comment by FlipperPA** [comment]
I agree with @browniebroke - and I think we can wait on enabling discussions. There's very little volume in issues, so I'd rather not have two places until there's significant volume that warrants a separation. Unless someone has a compelling reason otherwise! As to the Wiki, I prefer to keep everything in the README / docs for portability and to avoid lock-in.

**Comment by tim-schilling** [comment]
```
  "drf-excel" = {
    description          = "An XLSX spreadsheet renderer for Django REST Framework."{
    allow_merge_commit = true
    allow_rebase_merge = true
    allow_squash_merge = true
    allow_update_branch = true
    has_discussions = true
    enable_branch_protection = true # optional, default is true
    required_status_checks_contexts = [] # optional, default is []
    admins = [
        "FlipperPA",
    ]
    committers = []
    members = [
        "browniebroke",
    ]
  }
```

**Comment by cunla** [comment]
Welcome! Let us know if you have any questions.
